src/main/java/com/mealmentor/enterprise/MealMentorController.java

```diff
@@ -34,24 +33,30 @@ public String index() {
 
     @GetMapping("/mealItem")
     @ResponseBody
-    public List<MealItem> fetchAllMealItems(){
+    public List<MealItem> fetchAllMealItems() {
 
         return mealPlanService.fetchAll();
     }
 
     @GetMapping("/mealItem/{id}/")
     public ResponseEntity fetchMealItemById(@PathVariable("id") String id) {
-        MealItem foundMealItem = mealPlanService.fetchById(Integer.parseInt(id));
-        HttpHeaders headers = new HttpHeaders();
-        headers.setContentType(MediaType.APPLICATION_JSON);
-        return new ResponseEntity(foundMealItem, headers, HttpStatus.OK);
+        try {
+            MealItem foundMealItem = mealPlanService.fetchById(Integer.parseInt(id));
+            HttpHeaders headers = new HttpHeaders();
+            headers.setContentType(MediaType.APPLICATION_JSON);
+            return new ResponseEntity(foundMealItem, headers, HttpStatus.OK);
+        } catch(Exception e) {
+            e.printStackTrace();
+            return new ResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
```

Comment:
I added a try-catch block to capture any sort of exceptions that could be thrown by fetchById() method as it is calling the database to retrieve meal items. If something goes wrong, it will spit out a INTERNAL_SERVER_ERROR message to user instead of application crash

Here is a link that describes java best practices to handle exceptions: https://stackify.com/best-practices-exceptions-java/

Reply:
This is very good.  Excellent optimization on exception handling, and returning a known HTTP status code.

You can go one step further.  Use proper logging instead of printStackTrace().
- Logging has the concept of priority (error, debug, warn, etc.), which printStackTrace doesn't have.
- No one will see printStackTrace(), as it goes to console. 
- Log messages can be aggregated by several commercial tools.  Not so easy with printStackTrace.